codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -510,22 +511,26 @@ class JsonParserGenerator(
                                 """,
                                 *codegenScope,
                             )
-                            withBlock("variant = match key.to_unescaped()?.as_ref() {", "};") {
```

Comment:
What's the advantage to moving the `variant =` piece to inside of the match? Seems like it adds more overall code?

Reply:
Changed it back (it was used in the previous commit)